lib/archethic/self_repair/sync/transaction_handler.ex

```diff
@@ -66,25 +66,61 @@ defmodule Archethic.SelfRepair.Sync.TransactionHandler do
       |> Election.chain_storage_nodes_with_type(type, node_list)
       |> Enum.reject(&(&1.first_public_key == Crypto.first_node_public_key()))
 
+    download_nodes =
+      storage_nodes
+      |> P2P.filter_and_prioritize_nodes_for_quorum_read()
+
     case TransactionChain.fetch_transaction_remotely(address, storage_nodes) do
       {:ok, tx = %Transaction{}} ->
         tx
 
-      {:error, :transaction_not_exists} ->
-        Logger.error("Cannot fetch the transaction to sync",
-          transaction_address: Base.encode16(address),
-          transaction_type: type
-        )
-
-        raise "Transaction doesn't exist"
-
-      {:error, :network_issue} ->
-        Logger.error("Cannot fetch the transaction to sync",
-          transaction_address: Base.encode16(address),
-          transaction_type: type
-        )
+      {:error, error} ->
+        do_retry_download_transaction(address, type, download_nodes, error)
```

Comment:
Why not leverage `P2P.quorum_read` but extend it to support the return of the list of responding nodes, so you could discard them in the next iteration.
Hence, all the logic wouldn't need to be reimplemented.

Reply:
Good idea, if you think we will need to use this iteration process somewhere else.

Reply:
at first we were thinking that it is mostly needed here so there was no need to change the api behaviour